cmd/kyma/undeploy/cmd.go

```diff
@@ -197,3 +201,22 @@ func (cmd *command) initialSetup(wsp string) error {
 	preReqStep.Success()
 	return nil
 }
+
+// resolveVersion determines which version of kyma has to be used for undeploy.
+// Version is automatically detected from the cluster and can be optionally overridden via flag.
+func (cmd *command) resolveVersion() error {
+	if cmd.opts.Source != "" {
+		return nil
+	}
```

Comment:
If the version offered in cmd is not same as the version in the cluster, cluster version should win, or?
So this check should be here I think

Reply:
It is the other way around, version in the flag should override what is on the cluster. It is a bypass in case the version on the cluster is wrong or something like that.

Reply:
I'm a bit confused, how is the version can be wrong on the cluster? If the kyam get deployed, then the version either fixed or after upgrade, it can be changed, then do the undeploy, we should always take the latest version in cluster as source, and download kyma repo based on that version, or I understood it in wrong way?